myhpom/views/upload.py

```diff
@@ -35,6 +28,37 @@ def upload_current_ad(request):
     })
 
 
+@require_http_methods(['GET', 'POST'])
+@login_required
+def upload_requirements(request):
+    """
+    GET: show the upload/state_requirements form for the current user/state
+    POST: store the advance directive date, redirect to the upload/submit view.
+    """
+    MIN_YEAR = 1950
+    if request.method == "POST":
+        form = UploadRequirementsForm(request.POST)
+        if form.is_valid():
+            if hasattr(request.user, 'advancedirective'):
+                directive = request.user.advancedirective
+            else:
+                directive = AdvanceDirective(user=request.user, share_with_ehs=False)
+            directive.valid_date = form.cleaned_data['valid_date']
+            directive.save()
```

Comment:
Should not need to explicitly set `valid_date` here, the ModelForm should do that during `save()`.